File: 访问所有点的最小时间.js
var minTimeToVisitAllPoints = function(points) {
    let R=0;
    if(points.length == 0){
        return 0;
    }
    for(let i=0;i<points.length-1;i++){
        let X = Math.abs(points[i+1][0]-points[i][0]);//坐标X的差值
        let Y = Math.abs(points[i+1][1]-points[i][1]);//坐标Y的差值
        if(X>=Y){
            R = R+X;
        }else{
            R = R+Y; 
        }
    }
    return R
};

File: 访问所有点的最小时间.go
package main

func main(){
	
}

func minTimeToVisitAllPoints(points [][]int) int {
    if len(points) <= 1 {
        return 0
    }
    ret := 0
    for i := 1; i < len(points); i++ {
        ret +=  max(abs(points[i][0] - points[i-1][0]), abs(points[i][1] - points[i-1][1]))
    }
    return ret
}

func abs(x int) int {
    if x < 0 {
        return -1 * x
    }
    return x
}

func max(x, y int) int {
    if x >= y {
        return x
    }
    return y
}